app/main.py

```diff
@@ -1,3 +1,40 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km):
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
+            return Distance(self.km + other)
```

Comment:
When adding a non-Distance instance, it's assumed to be a numeric type. Consider adding a check to ensure `other` is a valid numeric type (int or float) to avoid unexpected behavior.